File: lib/src/test/java/com/hootsuite/nachos/ChipSpanChipCreatorTest.java
package com.hootsuite.nachos;

import android.content.Context;

import com.hootsuite.nachos.chip.ChipSpan;
import com.hootsuite.nachos.chip.ChipSpanChipCreator;

import junit.framework.TestCase;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(CustomRobolectricRunner.class)
@Config(constants = BuildConfig.class)
public class ChipSpanChipCreatorTest extends TestCase {

    private static final String SAMPLE_TEXT = "abcde";

    private Context mContext;
    private ChipSpanChipCreator mChipSpanChipCreator;

    @Before
    public void setup() {
        mContext = RuntimeEnvironment.application.getApplicationContext();
        mChipSpanChipCreator = new ChipSpanChipCreator();
    }

    @Test
    public void testCreateChip_fromText() {
        // run
        ChipSpan chipSpan = mChipSpanChipCreator.createChip(mContext, SAMPLE_TEXT, null);

        // verify
        assertThat(chipSpan).isNotNull();
        assertThat(chipSpan.getText().toString()).isEqualTo(SAMPLE_TEXT);
    }

    @Test
    public void testCreateChip_fromChip() {
        // setup
        ChipSpan existingChipSpan = new ChipSpan(mContext, SAMPLE_TEXT, null, null);

        // run
        ChipSpan chipSpan = mChipSpanChipCreator.createChip(mContext, existingChipSpan);

        // verify
        assertThat(chipSpan).isNotNull();
        assertThat(chipSpan.getText().toString()).isEqualTo(SAMPLE_TEXT);
        assertThat(chipSpan.getDrawable()).isNull();
    }

    @Test
    public void testCreateChip_fromTextWithData() {
        // run
        Object data = new Object();
        ChipSpan chipSpan = mChipSpanChipCreator.createChip(mContext, SAMPLE_TEXT, data);

        // verify
        assertThat(chipSpan).isNotNull();
        assertThat(chipSpan.getText().toString()).isEqualTo(SAMPLE_TEXT);
        assertThat(chipSpan.getData()).isSameAs(data);
    }

    @Test
    public void testCreateChip_fromChipWithData() {
        // setup
        Object data = new Object();
        ChipSpan existingChipSpan = new ChipSpan(mContext, SAMPLE_TEXT, null, data);

        // run
        ChipSpan chipSpan = mChipSpanChipCreator.createChip(mContext, existingChipSpan);

        // verify
        assertThat(chipSpan).isNotNull();
        assertThat(chipSpan.getText().toString()).isEqualTo(SAMPLE_TEXT);
        assertThat(chipSpan.getDrawable()).isNull();
        assertThat(chipSpan.getData()).isSameAs(data);
    }
}

File: nachos/src/test/java/com/hootsuite/nachos/ChipSpanChipCreatorTest.kt
package com.hootsuite.nachos

import android.content.Context
import com.hootsuite.nachos.chip.ChipSpan
import com.hootsuite.nachos.chip.ChipSpanChipCreator
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RuntimeEnvironment
import org.robolectric.annotation.Config

/**
 * Kotlin port of [ChipSpanChipCreatorTest]. Behaviour identical.
 */
@RunWith(CustomRobolectricRunner::class)
@Config(sdk = [TestConfig.SDK_VERSION])
class ChipSpanChipCreatorTest {

    private lateinit var context: Context
    private lateinit var chipCreator: ChipSpanChipCreator

    companion object {
        private const val SAMPLE_TEXT = "abcde"
    }

    @Before
    fun setUp() {
        context = RuntimeEnvironment.getApplication().applicationContext
        chipCreator = ChipSpanChipCreator()
    }

    @Test
    fun createChip_fromText() {
        val chipSpan: ChipSpan = chipCreator.createChip(context, SAMPLE_TEXT, null)
        assertThat(chipSpan).isNotNull
        assertThat(chipSpan.text.toString()).isEqualTo(SAMPLE_TEXT)
    }

    @Test
    fun createChip_fromChip() {
        val existing = ChipSpan(context, SAMPLE_TEXT, null, null)
        val chipSpan: ChipSpan = chipCreator.createChip(context, existing)
        assertThat(chipSpan).isNotNull
        assertThat(chipSpan.text.toString()).isEqualTo(SAMPLE_TEXT)
        assertThat(chipSpan.drawable).isNull()
    }

    @Test
    fun createChip_fromTextWithData() {
        val data = Any()
        val chipSpan: ChipSpan = chipCreator.createChip(context, SAMPLE_TEXT, data)
        assertThat(chipSpan).isNotNull
        assertThat(chipSpan.text.toString()).isEqualTo(SAMPLE_TEXT)
        assertThat(chipSpan.data).isSameAs(data)
    }

    @Test
    fun createChip_fromChipWithData() {
        val data = Any()
        val existing = ChipSpan(context, SAMPLE_TEXT, null, data)
        val chipSpan: ChipSpan = chipCreator.createChip(context, existing)
        assertThat(chipSpan).isNotNull
        assertThat(chipSpan.text.toString()).isEqualTo(SAMPLE_TEXT)
        assertThat(chipSpan.drawable).isNull()
        assertThat(chipSpan.data).isSameAs(data)
    }
} 